{"cells":[{"metadata":{},"cell_type":"markdown","source":"### Routine analysis for genome wide data using ***PLINK***"},{"metadata":{"trusted":true,"scrolled":true},"id":"embedded-personality","cell_type":"code","source":"%%bash\nunzip -o plinkfiles.zip","execution_count":1,"outputs":[{"output_type":"stream","text":"Archive:  plinkfiles.zip\n  inflating: GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.bed  \n  inflating: GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.bim  \n  inflating: GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.fam  \n  inflating: admixture               \n  inflating: admixture32             \n  inflating: ancient2merge_cleaned.bed  \n  inflating: ancient2merge_cleaned.bim  \n  inflating: ancient2merge_cleaned.fam  \n  inflating: ancient2merge_cleanedEurasia_nopug.bed  \n  inflating: ancient2merge_cleanedEurasia_nopug.bim  \n  inflating: ancient2merge_cleanedEurasia_nopug.fam  \n  inflating: ancient2merge_cleaned_nopug.bed  \n  inflating: ancient2merge_cleaned_nopug.bim  \n  inflating: ancient2merge_cleaned_nopug.fam  \n  inflating: modern2merge_cleanedEurasia_nopug.bed  \n  inflating: modern2merge_cleanedEurasia_nopug.bim  \n  inflating: modern2merge_cleanedEurasia_nopug.fam  \n  inflating: modern2merge_cleaned_nopug.bed  \n  inflating: modern2merge_cleaned_nopug.bim  \n  inflating: modern2merge_cleaned_nopug.fam  \n  inflating: plink                   \n","name":"stdout"}]},{"metadata":{},"cell_type":"markdown","source":"We now have a few different plink files we will use in this breif introduction. First, let's have a look again at the file we created in the ***Ex1***."},{"metadata":{"trusted":true},"id":"appointed-january","cell_type":"code","source":"%%bash \nhead GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.fam","execution_count":2,"outputs":[{"output_type":"stream","text":"PuBa GSM1094884 0 0 0 2\nPuBa GSM1094885 0 0 0 2\nPuBa GSM1094886 0 0 0 1\nPuBa GSM1094887 0 0 0 1\nPuBa GSM1094888 0 0 0 1\nPuBa GSM1094889 0 0 0 1\nPuBa GSM1094890 0 0 0 1\nPuBa GSM1094891 0 0 0 1\nPuBa GSM1094892 0 0 0 1\nPuBa GSM1094893 0 0 0 1\n","name":"stdout"}]},{"metadata":{},"cell_type":"markdown","source":"I do ***not** like the PuBa name for my population. I would better use a more explicit name such as ***Puglia***.\nOne way to replace name is using the bash command `sed`"},{"metadata":{"trusted":true},"cell_type":"code","source":"%%bash\nsed -e 's/PuBa/Puglia/g' GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.fam","execution_count":3,"outputs":[{"output_type":"stream","text":"Puglia GSM1094884 0 0 0 2\nPuglia GSM1094885 0 0 0 2\nPuglia GSM1094886 0 0 0 1\nPuglia GSM1094887 0 0 0 1\nPuglia GSM1094888 0 0 0 1\nPuglia GSM1094889 0 0 0 1\nPuglia GSM1094890 0 0 0 1\nPuglia GSM1094891 0 0 0 1\nPuglia GSM1094892 0 0 0 1\nPuglia GSM1094893 0 0 0 1\nPuglia GSM1094894 0 0 0 1\nPuglia GSM1094895 0 0 0 1\nPuglia GSM1094896 0 0 0 1\nPuglia GSM1094897 0 0 0 1\nPuglia GSM1094898 0 0 0 1\nPuglia GSM1094899 0 0 0 1\nPuglia GSM1094900 0 0 0 1\nPuglia GSM1094901 0 0 0 1\nPuglia GSM1094902 0 0 0 1\nPuglia GSM1094903 0 0 0 1\nPuglia GSM1094904 0 0 0 1\nPuglia GSM1094905 0 0 0 1\nPuglia GSM1094906 0 0 0 2\nPuglia GSM1094907 0 0 0 1\nPuglia GSM1094908 0 0 0 1\nPuglia GSM1094909 0 0 0 2\nPuglia GSM1094910 0 0 0 1\nPuglia GSM1094911 0 0 0 2\nPuglia GSM1094912 0 0 0 1\nPuglia GSM1094913 0 0 0 1\nPuglia GSM1094914 0 0 0 1\nPuglia GSM1094915 0 0 0 1\nPuglia GSM1094916 0 0 0 1\nPuglia GSM1094917 0 0 0 1\nPuglia GSM1094918 0 0 0 1\nPuglia GSM1094919 0 0 0 1\nPuglia GSM1094920 0 0 0 2\nPuglia GSM1094921 0 0 0 1\nPuglia GSM1094922 0 0 0 1\nPuglia GSM1094923 0 0 0 1\nPuglia GSM1094924 0 0 0 1\nPuglia GSM1094925 0 0 0 1\nPuglia GSM1094926 0 0 0 1\nPuglia GSM1094927 0 0 0 1\nPuglia GSM1094928 0 0 0 1\nPuglia GSM1094929 0 0 0 1\nPuglia GSM1094930 0 0 0 2\nPuglia GSM1094931 0 0 0 2\nPuglia GSM1094932 0 0 0 1\nPuglia GSM1094933 0 0 0 1\nPuglia GSM1094934 0 0 0 1\nPuglia GSM1094935 0 0 0 2\nPuglia GSM1094936 0 0 0 1\nPuglia GSM1094937 0 0 0 1\nPuglia GSM1094938 0 0 0 2\nPuglia GSM1094939 0 0 0 2\nPuglia GSM1094940 0 0 0 2\nPuglia GSM1094941 0 0 0 2\nPuglia GSM1094942 0 0 0 2\nPuglia GSM1094943 0 0 0 2\nPuglia GSM1094944 0 0 0 1\nPuglia GSM1094945 0 0 0 2\nPuglia GSM1094946 0 0 0 1\nPuglia GSM1094947 0 0 0 1\nPuglia GSM1094948 0 0 0 2\nPuglia GSM1094949 0 0 0 1\nPuglia GSM1094950 0 0 0 1\nPuglia GSM1094951 0 0 0 1\nPuglia GSM1094952 0 0 0 1\nPuglia GSM1094953 0 0 0 1\nPuglia GSM1094954 0 0 0 1\nPuglia GSM1094955 0 0 0 1\nPuglia GSM1094956 0 0 0 1\nPuglia GSM1094957 0 0 0 1\nPuglia GSM1094958 0 0 0 1\nPuglia GSM1094959 0 0 0 1\nPuglia GSM1094960 0 0 0 1\nPuglia GSM1094961 0 0 0 1\nPuglia GSM1094962 0 0 0 1\nPuglia GSM1094963 0 0 0 1\nPuglia GSM1094964 0 0 0 1\nPuglia GSM1094965 0 0 0 2\nPuglia GSM1094966 0 0 0 2\nPuglia GSM1094967 0 0 0 1\nPuglia GSM1094968 0 0 0 1\nPuglia GSM1094969 0 0 0 2\nPuglia GSM1094970 0 0 0 1\nPuglia GSM1094971 0 0 0 2\nPuglia GSM1094972 0 0 0 1\nPuglia GSM1094973 0 0 0 1\nPuglia GSM1094974 0 0 0 1\nPuglia GSM1094975 0 0 0 1\nPuglia GSM1094976 0 0 0 1\nPuglia GSM1094977 0 0 0 2\nPuglia GSM1094978 0 0 0 1\nPuglia GSM1094979 0 0 0 1\nPuglia GSM1094980 0 0 0 2\nPuglia GSM1094981 0 0 0 1\nPuglia GSM1094982 0 0 0 1\nPuglia GSM1094983 0 0 0 1\nPuglia GSM1094984 0 0 0 2\nPuglia GSM1094985 0 0 0 2\nPuglia GSM1094986 0 0 0 1\nPuglia GSM1094987 0 0 0 1\nPuglia GSM1094988 0 0 0 1\nPuglia GSM1094989 0 0 0 1\nPuglia GSM1094990 0 0 0 1\nPuglia GSM1094997 0 0 0 1\nPuglia GSM1094998 0 0 0 1\nPuglia GSM1094999 0 0 0 1\nPuglia GSM1095000 0 0 0 1\nPuglia GSM1095001 0 0 0 1\nPuglia GSM1095002 0 0 0 2\nPuglia GSM1095003 0 0 0 1\nPuglia GSM1095004 0 0 0 1\nPuglia GSM1095005 0 0 0 1\nPuglia GSM1095006 0 0 0 1\nPuglia GSM1095007 0 0 0 1\nPuglia GSM1095008 0 0 0 2\nPuglia GSM1095009 0 0 0 2\nPuglia GSM1095010 0 0 0 1\nPuglia GSM1095011 0 0 0 1\nPuglia GSM1095012 0 0 0 1\nPuglia GSM1095013 0 0 0 1\nPuglia GSM1095014 0 0 0 1\nPuglia GSM1095015 0 0 0 1\nPuglia GSM1095016 0 0 0 1\nPuglia GSM1095017 0 0 0 1\nPuglia GSM1095018 0 0 0 1\nPuglia GSM1095019 0 0 0 2\nPuglia GSM1095020 0 0 0 1\nPuglia GSM1095021 0 0 0 2\nPuglia GSM1095022 0 0 0 1\nPuglia GSM1095023 0 0 0 1\nPuglia GSM1095024 0 0 0 1\nPuglia GSM1095025 0 0 0 1\nPuglia GSM1095026 0 0 0 1\nPuglia GSM1095027 0 0 0 1\nPuglia GSM1095028 0 0 0 1\nPuglia GSM1095029 0 0 0 2\nPuglia GSM1095030 0 0 0 2\nPuglia GSM1095031 0 0 0 1\nPuglia GSM1095032 0 0 0 2\nPuglia GSM1095033 0 0 0 1\nPuglia GSM1095034 0 0 0 1\nPuglia GSM1095035 0 0 0 1\nPuglia GSM1095036 0 0 0 1\nPuglia GSM1095037 0 0 0 1\nPuglia GSM1095038 0 0 0 1\nPuglia GSM1095039 0 0 0 2\nPuglia GSM1095040 0 0 0 2\nPuglia GSM1095041 0 0 0 1\nPuglia GSM1095042 0 0 0 1\nPuglia GSM1095043 0 0 0 1\nPuglia GSM1095044 0 0 0 1\nPuglia GSM1095045 0 0 0 1\nPuglia GSM1095046 0 0 0 1\nPuglia GSM1095047 0 0 0 1\nPuglia GSM1095048 0 0 0 1\nPuglia GSM1095049 0 0 0 2\nPuglia GSM1095050 0 0 0 1\nPuglia GSM1095051 0 0 0 1\nPuglia GSM1095052 0 0 0 1\nPuglia GSM1095053 0 0 0 1\nPuglia GSM1095054 0 0 0 1\nPuglia GSM1095055 0 0 0 1\nPuglia GSM1095056 0 0 0 1\nPuglia GSM1095057 0 0 0 1\nPuglia GSM1095058 0 0 0 1\nPuglia GSM1095059 0 0 0 1\nPuglia GSM1095060 0 0 0 1\nPuglia GSM1095061 0 0 0 1\nPuglia GSM1095062 0 0 0 1\nPuglia GSM1095063 0 0 0 1\nPuglia GSM1095064 0 0 0 2\nPuglia GSM1095065 0 0 0 2\nPuglia GSM1095066 0 0 0 1\nPuglia GSM1095067 0 0 0 1\nPuglia GSM1095068 0 0 0 2\nPuglia GSM1095069 0 0 0 2\nPuglia GSM1095070 0 0 0 1\nPuglia GSM1095071 0 0 0 2\nPuglia GSM1095072 0 0 0 2\nPuglia GSM1095073 0 0 0 1\nPuglia GSM1095074 0 0 0 1\nPuglia GSM1095075 0 0 0 1\nPuglia GSM1095076 0 0 0 1\nPuglia GSM1095077 0 0 0 1\nPuglia GSM1095078 0 0 0 1\nPuglia GSM1095079 0 0 0 1\nPuglia GSM1095080 0 0 0 1\nPuglia GSM1095081 0 0 0 1\nPuglia GSM1095082 0 0 0 1\nPuglia GSM1095083 0 0 0 1\nPuglia GSM1095084 0 0 0 2\nPuglia GSM1095085 0 0 0 1\nPuglia GSM1095086 0 0 0 1\nPuglia GSM1095087 0 0 0 2\nPuglia GSM1095088 0 0 0 2\nPuglia GSM1095089 0 0 0 1\nPuglia GSM1095090 0 0 0 1\nPuglia GSM1095091 0 0 0 1\nPuglia GSM1095092 0 0 0 1\nPuglia GSM1095093 0 0 0 1\nPuglia GSM1095094 0 0 0 2\nPuglia GSM1095095 0 0 0 1\nPuglia GSM1095096 0 0 0 2\nPuglia GSM1095097 0 0 0 1\nPuglia GSM1095098 0 0 0 2\nPuglia GSM1095099 0 0 0 1\nPuglia GSM1095100 0 0 0 1\nPuglia GSM1095101 0 0 0 2\nPuglia GSM1095102 0 0 0 2\nPuglia GSM1095103 0 0 0 2\nPuglia GSM1095104 0 0 0 1\nPuglia GSM1095105 0 0 0 1\nPuglia GSM1095106 0 0 0 1\n","name":"stdout"}]},{"metadata":{"trusted":true},"cell_type":"code","source":"%%bash\nhead GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.fam","execution_count":4,"outputs":[{"output_type":"stream","text":"PuBa GSM1094884 0 0 0 2\nPuBa GSM1094885 0 0 0 2\nPuBa GSM1094886 0 0 0 1\nPuBa GSM1094887 0 0 0 1\nPuBa GSM1094888 0 0 0 1\nPuBa GSM1094889 0 0 0 1\nPuBa GSM1094890 0 0 0 1\nPuBa GSM1094891 0 0 0 1\nPuBa GSM1094892 0 0 0 1\nPuBa GSM1094893 0 0 0 1\n","name":"stdout"}]},{"metadata":{"trusted":true},"cell_type":"code","source":"%%bash\nsed -i 's/PuBa/Puglia/g' GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.fam","execution_count":5,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"%%bash\nhead GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.fam","execution_count":6,"outputs":[{"output_type":"stream","text":"Puglia GSM1094884 0 0 0 2\nPuglia GSM1094885 0 0 0 2\nPuglia GSM1094886 0 0 0 1\nPuglia GSM1094887 0 0 0 1\nPuglia GSM1094888 0 0 0 1\nPuglia GSM1094889 0 0 0 1\nPuglia GSM1094890 0 0 0 1\nPuglia GSM1094891 0 0 0 1\nPuglia GSM1094892 0 0 0 1\nPuglia GSM1094893 0 0 0 1\n","name":"stdout"}]},{"metadata":{},"cell_type":"markdown","source":"Most of the times, when carrying out genetic analysis, we want to explore the variability of target population in a broader context, therefore combining different populations. Let's have a look at the modern2merge_cleaned_nopug file. How many individuals are stored in it? How many markers? and populations?"},{"metadata":{"trusted":true,"scrolled":false},"id":"disabled-movement","cell_type":"code","source":"%%bash\n./plink --bfile GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD --bmerge modern2merge_cleaned_nopug --make-bed --out GreSalAndModern","execution_count":7,"outputs":[{"output_type":"stream","text":"PLINK v1.90b6.22 64-bit (16 Apr 2021)          www.cog-genomics.org/plink/1.9/\n(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\nLogging to GreSalAndModern.log.\nOptions in effect:\n  --bfile GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD\n  --bmerge modern2merge_cleaned_nopug\n  --make-bed\n  --out GreSalAndModern\n\n52331 MB RAM detected; reserving 26165 MB for main workspace.\n217 people loaded from\nGreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.fam.\n2292 people to be merged from modern2merge_cleaned_nopug.fam.\nOf these, 2292 are new, while 0 are present in the base dataset.\n125459 markers loaded from\nGreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.bim.\n93647 markers to be merged from modern2merge_cleaned_nopug.bim.\nOf these, 40909 are new, while 52738 are present in the base dataset.\n","name":"stdout"},{"output_type":"stream","text":"Error: 26506 variants with 3+ alleles present.\n* If you believe this is due to strand inconsistency, try --flip with\n  GreSalAndModern-merge.missnp.\n  (Warning: if this seems to work, strand errors involving SNPs with A/T or C/G\n  alleles probably remain in your data.  If LD between nearby SNPs is high,\n  --flip-scan should detect them.)\n* If you are dealing with genuine multiallelic variants, we recommend exporting\n  that subset of the data to VCF (via e.g. '--recode vcf'), merging with\n  another tool/script, and then importing the result; PLINK is not yet suited\n  to handling them.\nSee https://www.cog-genomics.org/plink/1.9/data#merge3 for more discussion.\n","name":"stderr"},{"output_type":"error","ename":"CalledProcessError","evalue":"Command 'b'./plink --bfile GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD --bmerge modern2merge_cleaned_nopug --make-bed --out GreSalAndModern\\n'' returned non-zero exit status 3.","traceback":["\u001b[0;31m---------------------------------------------------------------------------\u001b[0m","\u001b[0;31mCalledProcessError\u001b[0m                        Traceback (most recent call last)","\u001b[0;32m<ipython-input-7-a9e57e37df43>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mget_ipython\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrun_cell_magic\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'bash'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m''\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'./plink --bfile GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD --bmerge modern2merge_cleaned_nopug --make-bed --out GreSalAndModern\\n'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m","\u001b[0;32m/srv/conda/envs/notebook/lib/python3.7/site-packages/IPython/core/interactiveshell.py\u001b[0m in \u001b[0;36mrun_cell_magic\u001b[0;34m(self, magic_name, line, cell)\u001b[0m\n\u001b[1;32m   2389\u001b[0m             \u001b[0;32mwith\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbuiltin_trap\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2390\u001b[0m                 \u001b[0margs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mmagic_arg_s\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2391\u001b[0;31m                 \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2392\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2393\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m/srv/conda/envs/notebook/lib/python3.7/site-packages/IPython/core/magics/script.py\u001b[0m in \u001b[0;36mnamed_script_magic\u001b[0;34m(line, cell)\u001b[0m\n\u001b[1;32m    140\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    141\u001b[0m                 \u001b[0mline\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mscript\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 142\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshebang\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mline\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    143\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    144\u001b[0m         \u001b[0;31m# write a basic docstring:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m<decorator-gen-103>\u001b[0m in \u001b[0;36mshebang\u001b[0;34m(self, line, cell)\u001b[0m\n","\u001b[0;32m/srv/conda/envs/notebook/lib/python3.7/site-packages/IPython/core/magic.py\u001b[0m in \u001b[0;36m<lambda>\u001b[0;34m(f, *a, **k)\u001b[0m\n\u001b[1;32m    185\u001b[0m     \u001b[0;31m# but it's overkill for just that one bit of state.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    186\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mmagic_deco\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 187\u001b[0;31m         \u001b[0mcall\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mlambda\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    188\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    189\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mcallable\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m/srv/conda/envs/notebook/lib/python3.7/site-packages/IPython/core/magics/script.py\u001b[0m in \u001b[0;36mshebang\u001b[0;34m(self, line, cell)\u001b[0m\n\u001b[1;32m    243\u001b[0m             \u001b[0msys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstderr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mflush\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    244\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0margs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mraise_error\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreturncode\u001b[0m\u001b[0;34m!=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 245\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mCalledProcessError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreturncode\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moutput\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstderr\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0merr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    246\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    247\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_run_script\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mto_close\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;31mCalledProcessError\u001b[0m: Command 'b'./plink --bfile GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD --bmerge modern2merge_cleaned_nopug --make-bed --out GreSalAndModern\\n'' returned non-zero exit status 3."]}]},{"metadata":{},"cell_type":"markdown","source":"We got an error stating that there are 25k triallelic SNPs. This is a very high number, and it's very unlikely that it's happening. In fact, what we have here is that some SNP is coded differently, forward in one dataset and reverse in the other. For example, for some SNP we could have Allele1: C  and Allele2: T in the dataset 1, while Allele1: G and Allele2:A in dataset 2"},{"metadata":{"trusted":true},"cell_type":"code","source":"%%bash \nhead GreSalAndModern-merge.missnp > first10SNPs.txt\npaste \\\n<(egrep -f first10SNPs.txt GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD.bim) \\\n<(egrep -f first10SNPs.txt modern2merge_cleaned_nopug.bim)","execution_count":8,"outputs":[{"output_type":"stream","text":"10\t10_100016339\t0\t100016339\tA\tG\t10\t10_100016339\t127.082\t100016339\tT\tC\n10\t10_100036209\t0\t100036209\tA\tG\t10\t10_100036209\t127.133\t100036209\tT\tC\n10\t10_100046078\t0\t100046078\tA\tC\t10\t10_100046078\t127.169\t100046078\tT\tG\n10\t10_100077152\t0\t100077152\tA\tG\t10\t10_100077152\t127.287\t100077152\tT\tC\n10\t10_100087228\t0\t100087228\tA\tG\t10\t10_100087228\t127.399\t100087228\tT\tC\n10\t10_100088262\t0\t100088262\tG\tA\t10\t10_100088262\t127.4\t100088262\tC\tT\n10\t10_100101261\t0\t100101261\tG\tA\t10\t10_100101261\t127.415\t100101261\tC\tT\n10\t10_100111392\t0\t100111392\tG\tA\t10\t10_100111392\t127.46\t100111392\tC\tT\n10\t10_100218766\t0\t100218766\tG\tA\t10\t10_100218766\t127.615\t100218766\tC\tT\n10\t10_100229532\t0\t100229532\tA\tG\t10\t10_100229532\t127.641\t100229532\tT\tC\n","name":"stdout"}]},{"metadata":{},"cell_type":"markdown","source":"Let's follow the suggestion. Let's flip all the problematic SNPs in one dataset. Warning!!! If there are SNPs in the form C/G or A/T you should expect serious problems!"},{"metadata":{"trusted":true},"id":"hispanic-daily","cell_type":"code","source":"%%bash\n\n./plink --bfile GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD --flip GreSalAndModern-merge.missnp --make-bed --out GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLDFlipped","execution_count":9,"outputs":[{"output_type":"stream","text":"PLINK v1.90b6.22 64-bit (16 Apr 2021)          www.cog-genomics.org/plink/1.9/\n(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\nLogging to GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLDFlipped.log.\nOptions in effect:\n  --bfile GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLD\n  --flip GreSalAndModern-merge.missnp\n  --make-bed\n  --out GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLDFlipped\n\n52331 MB RAM detected; reserving 26165 MB for main workspace.\n125459 variants loaded from .bim file.\n217 people (0 males, 0 females, 217 ambiguous) loaded from .fam.\nAmbiguous sex IDs written to\nGreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLDFlipped.nosex\n.\n217 phenotype values loaded from .fam.\n--flip: 26506 SNPs flipped.\nUsing 1 thread (no multithreaded calculations invoked).\nBefore main variant filters, 217 founders and 0 nonfounders present.\nCalculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\nTotal genotyping rate is 0.996173.\n125459 variants and 217 people pass filters and QC.\nAmong remaining phenotypes, 51 are cases and 166 are controls.\n--make-bed to\nGreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLDFlipped.bed\n+\nGreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLDFlipped.bim\n+\nGreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLDFlipped.fam\n... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n","name":"stdout"},{"output_type":"stream","text":"Warning: Ignoring phenotypes of missing-sex samples.  If you don't want those\nphenotypes to be ignored, use the --allow-no-sex flag.\n","name":"stderr"}]},{"metadata":{},"cell_type":"markdown","source":"Ok, now we can merge it again!"},{"metadata":{"trusted":true},"id":"accessory-bathroom","cell_type":"code","source":"%%bash\n./plink --bfile GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLDFlipped \\\n--bmerge modern2merge_cleaned_nopug --make-bed --out GreSalAndModern --allow-no-sex","execution_count":16,"outputs":[{"output_type":"stream","text":"PLINK v1.90b6.22 64-bit (16 Apr 2021)          www.cog-genomics.org/plink/1.9/\n(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\nLogging to GreSalAndModern.log.\nOptions in effect:\n  --allow-no-sex\n  --bfile GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLDFlipped\n  --bmerge modern2merge_cleaned_nopug\n  --make-bed\n  --out GreSalAndModern\n\n52331 MB RAM detected; reserving 26165 MB for main workspace.\n217 people loaded from\nGreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLDFlipped.fam.\n2292 people to be merged from modern2merge_cleaned_nopug.fam.\nOf these, 2292 are new, while 0 are present in the base dataset.\n125459 markers loaded from\nGreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLDFlipped.bim.\n93647 markers to be merged from modern2merge_cleaned_nopug.bim.\nOf these, 40909 are new, while 52738 are present in the base dataset.\nPerforming single-pass merge (2509 people, 166368 variants).\n\rPass 1: fileset #1 complete.\r                                              \rMerged fileset written to GreSalAndModern-merge.bed + GreSalAndModern-merge.bim\n+ GreSalAndModern-merge.fam .\n166368 variants loaded from .bim file.\n2509 people (1366 males, 434 females, 709 ambiguous) loaded from .fam.\nAmbiguous sex IDs written to GreSalAndModern.nosex .\n485 phenotype values loaded from .fam.\nUsing 1 thread (no multithreaded calculations invoked).\nBefore main variant filters, 2509 founders and 0 nonfounders present.\nCalculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\nTotal genotyping rate is 0.578476.\n166368 variants and 2509 people pass filters and QC.\nAmong remaining phenotypes, 51 are cases and 434 are controls.  (2024\nphenotypes are missing.)\n--make-bed to GreSalAndModern.bed + GreSalAndModern.bim + GreSalAndModern.fam\n... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n","name":"stdout"}]},{"metadata":{"trusted":true},"id":"legendary-dancing","cell_type":"code","source":"%%bash\n./plink --bfile GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLDFlipped \\\n--bmerge modern2merge_cleanedEurasia_nopug --make-bed --out GreSalAndModernEurasia --allow-no-sex","execution_count":17,"outputs":[{"output_type":"stream","text":"PLINK v1.90b6.22 64-bit (16 Apr 2021)          www.cog-genomics.org/plink/1.9/\n(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\nLogging to GreSalAndModernEurasia.log.\nOptions in effect:\n  --allow-no-sex\n  --bfile GreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLDFlipped\n  --bmerge modern2merge_cleanedEurasia_nopug\n  --make-bed\n  --out GreSalAndModernEurasia\n\n52331 MB RAM detected; reserving 26165 MB for main workspace.\n217 people loaded from\nGreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLDFlipped.fam.\n1014 people to be merged from modern2merge_cleanedEurasia_nopug.fam.\nOf these, 1014 are new, while 0 are present in the base dataset.\n125459 markers loaded from\nGreSalVL_b37_AllelesUpdatedNewMapBEDAutoGeno05Mind05ControlsHWERelativeRemovedLDFlipped.bim.\n93647 markers to be merged from modern2merge_cleanedEurasia_nopug.bim.\nOf these, 40909 are new, while 52738 are present in the base dataset.\nPerforming single-pass merge (1231 people, 166368 variants).\n\rPass 1: fileset #1 complete.\r                                              \rMerged fileset written to GreSalAndModernEurasia-merge.bed +\nGreSalAndModernEurasia-merge.bim + GreSalAndModernEurasia-merge.fam .\n166368 variants loaded from .bim file.\n1231 people (654 males, 300 females, 277 ambiguous) loaded from .fam.\nAmbiguous sex IDs written to GreSalAndModernEurasia.nosex .\n476 phenotype values loaded from .fam.\nUsing 1 thread (no multithreaded calculations invoked).\nBefore main variant filters, 1231 founders and 0 nonfounders present.\nCalculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\nTotal genotyping rate is 0.59559.\n166368 variants and 1231 people pass filters and QC.\nAmong remaining phenotypes, 51 are cases and 425 are controls.  (755 phenotypes\nare missing.)\n--make-bed to GreSalAndModernEurasia.bed + GreSalAndModernEurasia.bim +\nGreSalAndModernEurasia.fam ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n","name":"stdout"}]},{"metadata":{"trusted":true},"id":"august-entertainment","cell_type":"code","source":"%%bash\n./plink --bfile GreSalAndModern --geno .05 --make-bed --out GreSalAndModern\n./plink --bfile GreSalAndModern --mind .05 --make-bed --out GreSalAndModern\n./plink --bfile GreSalAndModernEurasia --geno .05 --make-bed --out GreSalAndModernEurasia\n./plink --bfile GreSalAndModernEurasia --mind .05 --make-bed --out GreSalAndModernEurasia","execution_count":19,"outputs":[{"output_type":"stream","text":"PLINK v1.90b6.22 64-bit (16 Apr 2021)          www.cog-genomics.org/plink/1.9/\n(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\nLogging to GreSalAndModern.log.\nOptions in effect:\n  --bfile GreSalAndModern\n  --geno .05\n  --make-bed\n  --out GreSalAndModern\n\n52331 MB RAM detected; reserving 26165 MB for main workspace.\n52736 variants loaded from .bim file.\n2509 people (1366 males, 434 females, 709 ambiguous) loaded from .fam.\nAmbiguous sex IDs written to GreSalAndModern.nosex .\n485 phenotype values loaded from .fam.\nUsing 1 thread (no multithreaded calculations invoked).\nBefore main variant filters, 2509 founders and 0 nonfounders present.\nCalculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\nTotal genotyping rate is 0.998539.\n0 variants removed due to missing genotype data (--geno).\n52736 variants and 2509 people pass filters and QC.\nAmong remaining phenotypes, 51 are cases and 434 are controls.  (2024\nphenotypes are missing.)\n--make-bed to GreSalAndModern.bed + GreSalAndModern.bim + GreSalAndModern.fam\n... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\nPLINK v1.90b6.22 64-bit (16 Apr 2021)          www.cog-genomics.org/plink/1.9/\n(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\nLogging to GreSalAndModern.log.\nOptions in effect:\n  --bfile GreSalAndModern\n  --make-bed\n  --mind .05\n  --out GreSalAndModern\n\n52331 MB RAM detected; reserving 26165 MB for main workspace.\n52736 variants loaded from .bim file.\n2509 people (1366 males, 434 females, 709 ambiguous) loaded from .fam.\nAmbiguous sex IDs written to GreSalAndModern.nosex .\n485 phenotype values loaded from .fam.\n2 people removed due to missing genotype data (--mind).\nIDs written to GreSalAndModern.irem .\nUsing 1 thread (no multithreaded calculations invoked).\nBefore main variant filters, 2507 founders and 0 nonfounders present.\nCalculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\nTotal genotyping rate in remaining samples is 0.998706.\n52736 variants and 2507 people pass filters and QC.\nAmong remaining phenotypes, 51 are cases and 432 are controls.  (2024\nphenotypes are missing.)\n--make-bed to GreSalAndModern.bed + GreSalAndModern.bim + GreSalAndModern.fam\n... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\nPLINK v1.90b6.22 64-bit (16 Apr 2021)          www.cog-genomics.org/plink/1.9/\n(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\nLogging to GreSalAndModernEurasia.log.\nOptions in effect:\n  --bfile GreSalAndModernEurasia\n  --geno .05\n  --make-bed\n  --out GreSalAndModernEurasia\n\n52331 MB RAM detected; reserving 26165 MB for main workspace.\n166368 variants loaded from .bim file.\n1231 people (654 males, 300 females, 277 ambiguous) loaded from .fam.\nAmbiguous sex IDs written to GreSalAndModernEurasia.nosex .\n476 phenotype values loaded from .fam.\nUsing 1 thread (no multithreaded calculations invoked).\nBefore main variant filters, 1231 founders and 0 nonfounders present.\nCalculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\nTotal genotyping rate is 0.59559.\n113640 variants removed due to missing genotype data (--geno).\n52728 variants and 1231 people pass filters and QC.\nAmong remaining phenotypes, 51 are cases and 425 are controls.  (755 phenotypes\nare missing.)\n--make-bed to GreSalAndModernEurasia.bed + GreSalAndModernEurasia.bim +\nGreSalAndModernEurasia.fam ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\nPLINK v1.90b6.22 64-bit (16 Apr 2021)          www.cog-genomics.org/plink/1.9/\n(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\nLogging to GreSalAndModernEurasia.log.\nOptions in effect:\n  --bfile GreSalAndModernEurasia\n  --make-bed\n  --mind .05\n  --out GreSalAndModernEurasia\n\n52331 MB RAM detected; reserving 26165 MB for main workspace.\n52728 variants loaded from .bim file.\n1231 people (654 males, 300 females, 277 ambiguous) loaded from .fam.\nAmbiguous sex IDs written to GreSalAndModernEurasia.nosex .\n476 phenotype values loaded from .fam.\n2 people removed due to missing genotype data (--mind).\nIDs written to GreSalAndModernEurasia.irem .\nUsing 1 thread (no multithreaded calculations invoked).\nBefore main variant filters, 1229 founders and 0 nonfounders present.\nCalculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\nTotal genotyping rate in remaining samples is 0.99897.\n52728 variants and 1229 people pass filters and QC.\nAmong remaining phenotypes, 51 are cases and 423 are controls.  (755 phenotypes\nare missing.)\n--make-bed to GreSalAndModernEurasia.bed + GreSalAndModernEurasia.bim +\nGreSalAndModernEurasia.fam ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n","name":"stdout"},{"output_type":"stream","text":"Warning: --make-bed input and output filenames match.  Appending '~' to input\nfilenames.\nWarning: Ignoring phenotypes of missing-sex samples.  If you don't want those\nphenotypes to be ignored, use the --allow-no-sex flag.\nWarning: --make-bed input and output filenames match.  Appending '~' to input\nfilenames.\nWarning: Ignoring phenotypes of missing-sex samples.  If you don't want those\nphenotypes to be ignored, use the --allow-no-sex flag.\nWarning: --make-bed input and output filenames match.  Appending '~' to input\nfilenames.\nWarning: Ignoring phenotypes of missing-sex samples.  If you don't want those\nphenotypes to be ignored, use the --allow-no-sex flag.\nWarning: --make-bed input and output filenames match.  Appending '~' to input\nfilenames.\nWarning: Ignoring phenotypes of missing-sex samples.  If you don't want those\nphenotypes to be ignored, use the --allow-no-sex flag.\n","name":"stderr"}]},{"metadata":{"trusted":true},"id":"alpha-harris","cell_type":"code","source":"%%bash\n./plink --bfile GreSalAndModern --pca 4 --out GreSalAndModernPCA\n./plink --bfile GreSalAndModernEurasia --pca 4 --out GreSalAndModernEurasiaPCA","execution_count":null,"outputs":[]},{"metadata":{"trusted":true},"id":"latin-style","cell_type":"code","source":"%%bash\n./plink --bfile GreSalAndModern --mds-plot 2 --cluster --out GreSalAndModernMDS\n./plink --bfile GreSalAndModernEurasia --mds-plot 2 --cluster --out GreSalAndModernEurasiaMDS","execution_count":null,"outputs":[]},{"metadata":{"trusted":true},"id":"individual-probability","cell_type":"code","source":"%%bash\n./admixture --help","execution_count":null,"outputs":[]},{"metadata":{"trusted":false},"id":"surface-universe","cell_type":"code","source":"%%bash \n./plink --bfile GreSalAndModern --thin-count 10000 --make-bed --out GreSalAndModern10kSnps\n./plink --bfile GreSalAndModernEurasia --extract GreSalAndModern10kSnps.bim --make-bed --out GreSalAndModernEurasia10kSnps","execution_count":13,"outputs":[{"name":"stdout","output_type":"stream","text":"PLINK v1.90b6.22 64-bit (16 Apr 2021)          www.cog-genomics.org/plink/1.9/\n(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\nLogging to GreSalAndModern10kSnps.log.\nOptions in effect:\n  --bfile GreSalAndModern\n  --make-bed\n  --out GreSalAndModern10kSnps\n  --thin-count 10000\n\n9409 MB RAM detected; reserving 4704 MB for main workspace.\n93647 variants loaded from .bim file.\n2509 people (1366 males, 434 females, 709 ambiguous) loaded from .fam.\nAmbiguous sex IDs written to GreSalAndModern10kSnps.nosex .\n485 phenotype values loaded from .fam.\n--thin-count: 83647 variants removed (10000 remaining).\nUsing 1 thread (no multithreaded calculations invoked).\nBefore main variant filters, 2509 founders and 0 nonfounders present.\nCalculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\nTotal genotyping rate is 0.961655.\n10000 variants and 2509 people pass filters and QC.\nAmong remaining phenotypes, 51 are cases and 434 are controls.  (2024\nphenotypes are missing.)\n--make-bed to GreSalAndModern10kSnps.bed + GreSalAndModern10kSnps.bim +\nGreSalAndModern10kSnps.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\nPLINK v1.90b6.22 64-bit (16 Apr 2021)          www.cog-genomics.org/plink/1.9/\n(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\nLogging to GreSalAndModernEurasia10kSnps.log.\nOptions in effect:\n  --bfile GreSalAndModernEurasia\n  --extract GreSalAndModern10kSnps.bim\n  --make-bed\n  --out GreSalAndModernEurasia10kSnps\n\n9409 MB RAM detected; reserving 4704 MB for main workspace.\n93647 variants loaded from .bim file.\n1231 people (654 males, 300 females, 277 ambiguous) loaded from .fam.\nAmbiguous sex IDs written to GreSalAndModernEurasia10kSnps.nosex .\n476 phenotype values loaded from .fam.\n--extract: 10000 variants remaining.\nUsing 1 thread (no multithreaded calculations invoked).\nBefore main variant filters, 1231 founders and 0 nonfounders present.\nCalculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\nTotal genotyping rate is 0.923441.\n10000 variants and 1231 people pass filters and QC.\nAmong remaining phenotypes, 51 are cases and 425 are controls.  (755 phenotypes\nare missing.)\n--make-bed to GreSalAndModernEurasia10kSnps.bed +\nGreSalAndModernEurasia10kSnps.bim + GreSalAndModernEurasia10kSnps.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"},{"name":"stderr","output_type":"stream","text":"Warning: Ignoring phenotypes of missing-sex samples.  If you don't want those\nphenotypes to be ignored, use the --allow-no-sex flag.\nWarning: Ignoring phenotypes of missing-sex samples.  If you don't want those\nphenotypes to be ignored, use the --allow-no-sex flag.\n"}]},{"metadata":{"trusted":false},"id":"advance-hobby","cell_type":"code","source":"%%bash\nwc -l GreSalAndModern10kSnps.fam\nwc -l GreSalAndModernEurasia10kSnps.fam","execution_count":14,"outputs":[{"name":"stdout","output_type":"stream","text":"2509 GreSalAndModern10kSnps.fam\n1231 GreSalAndModernEurasia10kSnps.fam\n"}]},{"metadata":{"trusted":false},"id":"moved-barbados","cell_type":"code","source":"%%bash\nfor K in 2\ndo\necho processing $K\n./admixture GreSalAndModern10kSnps.bed $K\n./admixture GreSalAndModernEurasia10kSnps.bed $K\ndone","execution_count":15,"outputs":[{"name":"stdout","output_type":"stream","text":"processing 2\n****                   ADMIXTURE Version 1.3.0                  ****\n****                    Copyright 2008-2015                     ****\n****           David Alexander, Suyash Shringarpure,            ****\n****                John  Novembre, Ken Lange                   ****\n****                                                            ****\n****                 Please cite our paper!                     ****\n****   Information at www.genetics.ucla.edu/software/admixture  ****\n\nRandom seed: 43\nPoint estimation method: Block relaxation algorithm\nConvergence acceleration algorithm: QuasiNewton, 3 secant conditions\nPoint estimation will terminate when objective function delta < 0.0001\nEstimation of standard errors disabled; will compute point estimates only.\nSize of G: 2509x10000\nPerforming five EM steps to prime main algorithm\n1 (EM) \tElapsed: 4.48\tLoglikelihood: -2.72484e+07\t(delta): 1.42068e+07\n2 (EM) \tElapsed: 3.275\tLoglikelihood: -2.6544e+07\t(delta): 704391\n3 (EM) \tElapsed: 2.889\tLoglikelihood: -2.63359e+07\t(delta): 208038\n4 (EM) \tElapsed: 2.661\tLoglikelihood: -2.62337e+07\t(delta): 102190\n5 (EM) \tElapsed: 2.488\tLoglikelihood: -2.61787e+07\t(delta): 55011.4\nInitial loglikelihood: -2.61787e+07\nStarting main algorithm\n1 (QN/Block) \tElapsed: 8.342\tLoglikelihood: -2.57653e+07\t(delta): 413467\n2 (QN/Block) \tElapsed: 6.703\tLoglikelihood: -2.56292e+07\t(delta): 136085\n3 (QN/Block) \tElapsed: 7.169\tLoglikelihood: -2.55738e+07\t(delta): 55383.3\n4 (QN/Block) \tElapsed: 10.623\tLoglikelihood: -2.5534e+07\t(delta): 39817.5\n5 (QN/Block) \tElapsed: 11.38\tLoglikelihood: -2.55304e+07\t(delta): 3535.86\n6 (QN/Block) \tElapsed: 9.76\tLoglikelihood: -2.55298e+07\t(delta): 652.091\n7 (QN/Block) \tElapsed: 8.056\tLoglikelihood: -2.55295e+07\t(delta): 270.995\n8 (QN/Block) \tElapsed: 8.563\tLoglikelihood: -2.55293e+07\t(delta): 170.448\n9 (QN/Block) \tElapsed: 7.649\tLoglikelihood: -2.55292e+07\t(delta): 95.4182\n10 (QN/Block) \tElapsed: 7.714\tLoglikelihood: -2.55292e+07\t(delta): 57.3997\n11 (QN/Block) \tElapsed: 7.572\tLoglikelihood: -2.55292e+07\t(delta): 11.8796\n12 (QN/Block) \tElapsed: 7.605\tLoglikelihood: -2.55292e+07\t(delta): 11.7077\n13 (QN/Block) \tElapsed: 7.427\tLoglikelihood: -2.55292e+07\t(delta): 2.5488\n14 (QN/Block) \tElapsed: 9.662\tLoglikelihood: -2.55292e+07\t(delta): 0.0690086\n15 (QN/Block) \tElapsed: 7.98\tLoglikelihood: -2.55292e+07\t(delta): 0.0346305\n16 (QN/Block) \tElapsed: 7.723\tLoglikelihood: -2.55292e+07\t(delta): 0.00194236\n17 (QN/Block) \tElapsed: 8.041\tLoglikelihood: -2.55292e+07\t(delta): 2.78428e-05\nSummary: \nConverged in 17 iterations (172.112 sec)\nLoglikelihood: -25529166.044628\nFst divergences between estimated populations: \n\tPop0\t\nPop0\t\nPop1\t0.132\t\nWriting output files.\n****                   ADMIXTURE Version 1.3.0                  ****\n****                    Copyright 2008-2015                     ****\n****           David Alexander, Suyash Shringarpure,            ****\n****                John  Novembre, Ken Lange                   ****\n****                                                            ****\n****                 Please cite our paper!                     ****\n****   Information at www.genetics.ucla.edu/software/admixture  ****\n\nRandom seed: 43\nPoint estimation method: Block relaxation algorithm\nConvergence acceleration algorithm: QuasiNewton, 3 secant conditions\nPoint estimation will terminate when objective function delta < 0.0001\nEstimation of standard errors disabled; will compute point estimates only.\nSize of G: 1231x10000\nPerforming five EM steps to prime main algorithm\n1 (EM) \tElapsed: 1.208\tLoglikelihood: -1.24911e+07\t(delta): 7.03371e+06\n2 (EM) \tElapsed: 1.138\tLoglikelihood: -1.21659e+07\t(delta): 325181\n3 (EM) \tElapsed: 1.121\tLoglikelihood: -1.20692e+07\t(delta): 96656\n4 (EM) \tElapsed: 1.113\tLoglikelihood: -1.20223e+07\t(delta): 46942.7\n5 (EM) \tElapsed: 1.095\tLoglikelihood: -1.19979e+07\t(delta): 24333.7\nInitial loglikelihood: -1.19979e+07\nStarting main algorithm\n1 (QN/Block) \tElapsed: 2.92\tLoglikelihood: -1.1966e+07\t(delta): 31932.7\n2 (QN/Block) \tElapsed: 2.903\tLoglikelihood: -1.1953e+07\t(delta): 13025.3\n3 (QN/Block) \tElapsed: 3.136\tLoglikelihood: -1.19477e+07\t(delta): 5322.97\n4 (QN/Block) \tElapsed: 4.108\tLoglikelihood: -1.19439e+07\t(delta): 3788.47\n5 (QN/Block) \tElapsed: 3.922\tLoglikelihood: -1.19419e+07\t(delta): 1969.06\n6 (QN/Block) \tElapsed: 3.976\tLoglikelihood: -1.19406e+07\t(delta): 1256.12\n7 (QN/Block) \tElapsed: 3.934\tLoglikelihood: -1.19393e+07\t(delta): 1341.82\n8 (QN/Block) \tElapsed: 3.925\tLoglikelihood: -1.1938e+07\t(delta): 1335.77\n9 (QN/Block) \tElapsed: 3.905\tLoglikelihood: -1.19368e+07\t(delta): 1130.64\n10 (QN/Block) \tElapsed: 3.657\tLoglikelihood: -1.1936e+07\t(delta): 796.21\n11 (QN/Block) \tElapsed: 3.969\tLoglikelihood: -1.19356e+07\t(delta): 444.055\n12 (QN/Block) \tElapsed: 4.004\tLoglikelihood: -1.19354e+07\t(delta): 161.951\n13 (QN/Block) \tElapsed: 3.553\tLoglikelihood: -1.19353e+07\t(delta): 185.862\n14 (QN/Block) \tElapsed: 3.914\tLoglikelihood: -1.1935e+07\t(delta): 210.806\n15 (QN/Block) \tElapsed: 3.973\tLoglikelihood: -1.1934e+07\t(delta): 1045.93\n16 (QN/Block) \tElapsed: 4.186\tLoglikelihood: -1.19335e+07\t(delta): 452.51\n17 (QN/Block) \tElapsed: 3.857\tLoglikelihood: -1.19333e+07\t(delta): 243.589\n18 (QN/Block) \tElapsed: 3.578\tLoglikelihood: -1.19331e+07\t(delta): 166.538\n19 (QN/Block) \tElapsed: 3.19\tLoglikelihood: -1.1933e+07\t(delta): 113.866\n20 (QN/Block) \tElapsed: 3.173\tLoglikelihood: -1.19329e+07\t(delta): 110.973\n21 (QN/Block) \tElapsed: 3.183\tLoglikelihood: -1.19328e+07\t(delta): 98.0428\n22 (QN/Block) \tElapsed: 3.153\tLoglikelihood: -1.19324e+07\t(delta): 420.509\n23 (QN/Block) \tElapsed: 3.496\tLoglikelihood: -1.19321e+07\t(delta): 267.389\n24 (QN/Block) \tElapsed: 4.015\tLoglikelihood: -1.19319e+07\t(delta): 204.661\n25 (QN/Block) \tElapsed: 3.829\tLoglikelihood: -1.19318e+07\t(delta): 138.124\n26 (QN/Block) \tElapsed: 3.884\tLoglikelihood: -1.19316e+07\t(delta): 183.935\n27 (QN/Block) \tElapsed: 3.458\tLoglikelihood: -1.19314e+07\t(delta): 166.542\n28 (QN/Block) \tElapsed: 3.205\tLoglikelihood: -1.19313e+07\t(delta): 125.947\n29 (QN/Block) \tElapsed: 3.156\tLoglikelihood: -1.19313e+07\t(delta): 45.9697\n30 (QN/Block) \tElapsed: 3.132\tLoglikelihood: -1.19311e+07\t(delta): 137.662\n31 (QN/Block) \tElapsed: 3.184\tLoglikelihood: -1.1931e+07\t(delta): 97.5824\n32 (QN/Block) \tElapsed: 3.132\tLoglikelihood: -1.19309e+07\t(delta): 76.681\n33 (QN/Block) \tElapsed: 3.131\tLoglikelihood: -1.19309e+07\t(delta): 21.1028\n34 (QN/Block) \tElapsed: 3.14\tLoglikelihood: -1.19309e+07\t(delta): 39.409\n35 (QN/Block) \tElapsed: 3.161\tLoglikelihood: -1.19308e+07\t(delta): 63.1965\n36 (QN/Block) \tElapsed: 3.162\tLoglikelihood: -1.19308e+07\t(delta): 45.511\n37 (QN/Block) \tElapsed: 3.144\tLoglikelihood: -1.19308e+07\t(delta): 11.7354\n38 (QN/Block) \tElapsed: 3.17\tLoglikelihood: -1.19307e+07\t(delta): 43.9858\n39 (QN/Block) \tElapsed: 3.151\tLoglikelihood: -1.19307e+07\t(delta): 31.9256\n40 (QN/Block) \tElapsed: 3.234\tLoglikelihood: -1.19307e+07\t(delta): 18.8856\n41 (QN/Block) \tElapsed: 3.181\tLoglikelihood: -1.19307e+07\t(delta): 4.20376\n42 (QN/Block) \tElapsed: 3.129\tLoglikelihood: -1.19306e+07\t(delta): 23.5204\n43 (QN/Block) \tElapsed: 3.126\tLoglikelihood: -1.19306e+07\t(delta): 9.42687\n44 (QN/Block) \tElapsed: 3.158\tLoglikelihood: -1.19306e+07\t(delta): 3.9634\n45 (QN/Block) \tElapsed: 3.247\tLoglikelihood: -1.19306e+07\t(delta): 0.757438\n46 (QN/Block) \tElapsed: 3.121\tLoglikelihood: -1.19306e+07\t(delta): 0.351273\n47 (QN/Block) \tElapsed: 3.144\tLoglikelihood: -1.19306e+07\t(delta): 0.536069\n48 (QN/Block) \tElapsed: 3.607\tLoglikelihood: -1.19306e+07\t(delta): 0.0694177\n49 (QN/Block) \tElapsed: 3.706\tLoglikelihood: -1.19306e+07\t(delta): 0.00637329\n50 (QN/Block) \tElapsed: 3.371\tLoglikelihood: -1.19306e+07\t(delta): 0.00268759\n51 (QN/Block) \tElapsed: 3.254\tLoglikelihood: -1.19306e+07\t(delta): 0.00385649\n52 (QN/Block) \tElapsed: 3.225\tLoglikelihood: -1.19306e+07\t(delta): 0.000556698\n53 (QN/Block) \tElapsed: 3.223\tLoglikelihood: -1.19306e+07\t(delta): 5.90142e-05\nSummary: \nConverged in 53 iterations (204.385 sec)\nLoglikelihood: -11930625.390384\nFst divergences between estimated populations: \n\tPop0\t\nPop0\t\nPop1\t0.143\t\nWriting output files.\n"}]},{"metadata":{"trusted":false},"id":"supreme-longitude","cell_type":"code","source":"%%bash \n./plink --bfile ancient2merge_cleaned_nopug --extract GreSalAndModern10kSnps.bim --make-bed --out ancient2merge_cleaned_nopug10KSnps","execution_count":16,"outputs":[{"name":"stdout","output_type":"stream","text":"PLINK v1.90b6.22 64-bit (16 Apr 2021)          www.cog-genomics.org/plink/1.9/\n(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\nLogging to ancient2merge_cleaned_nopug10KSnps.log.\nOptions in effect:\n  --bfile ancient2merge_cleaned_nopug\n  --extract GreSalAndModern10kSnps.bim\n  --make-bed\n  --out ancient2merge_cleaned_nopug10KSnps\n\n9409 MB RAM detected; reserving 4704 MB for main workspace.\n93647 variants loaded from .bim file.\n92 people (51 males, 41 females) loaded from .fam.\n92 phenotype values loaded from .fam.\n--extract: 10000 variants remaining.\nUsing 1 thread (no multithreaded calculations invoked).\nBefore main variant filters, 92 founders and 0 nonfounders present.\nCalculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\nTotal genotyping rate is 0.644013.\n10000 variants and 92 people pass filters and QC.\nAmong remaining phenotypes, 0 are cases and 92 are controls.\n--make-bed to ancient2merge_cleaned_nopug10KSnps.bed +\nancient2merge_cleaned_nopug10KSnps.bim + ancient2merge_cleaned_nopug10KSnps.fam\n... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"}]},{"metadata":{"trusted":false},"id":"baking-parameter","cell_type":"code","source":"%%bash\ncp GreSalAndModern10kSnps.2.P ancient2merge_cleaned_nopug10KSnps.2.P.in\ncp GreSalAndModernEurasia10kSnps.2.P ancient2merge_cleaned_nopug10KSnps.2.P.in","execution_count":17,"outputs":[]},{"metadata":{"trusted":false},"id":"black-visibility","cell_type":"code","source":"%%bash\nwc -l ancient2merge_cleaned_nopug10KSnps.bim","execution_count":18,"outputs":[{"name":"stdout","output_type":"stream","text":"10000 ancient2merge_cleaned_nopug10KSnps.bim\n"}]},{"metadata":{"trusted":false},"id":"western-technician","cell_type":"code","source":"%%bash \n./admixture -P ancient2merge_cleaned_nopug10KSnps.bed 2","execution_count":19,"outputs":[{"name":"stdout","output_type":"stream","text":"****                   ADMIXTURE Version 1.3.0                  ****\n****                    Copyright 2008-2015                     ****\n****           David Alexander, Suyash Shringarpure,            ****\n****                John  Novembre, Ken Lange                   ****\n****                                                            ****\n****                 Please cite our paper!                     ****\n****   Information at www.genetics.ucla.edu/software/admixture  ****\n\nWill read fixed P from files\nRandom seed: 43\nPoint estimation method: Block relaxation algorithm\nP parameters will be fixed at initial values.\nConvergence acceleration algorithm: QuasiNewton, 3 secant conditions\nPoint estimation will terminate when objective function delta < 0.0001\nEstimation of standard errors disabled; will compute point estimates only.\nInitializing parameter matrices from input files:\n  Read P input from ancient2merge_cleaned_nopug10KSnps.2.P.in\nError: detected that all genotypes are missing for a SNP locus.\nPlease apply quality-control filters to remove such loci.\n"},{"ename":"CalledProcessError","evalue":"Command 'b'./admixture -P ancient2merge_cleaned_nopug10KSnps.bed 2\\n'' returned non-zero exit status 255.","output_type":"error","traceback":["\u001b[0;31m---------------------------------------------------------------------------\u001b[0m","\u001b[0;31mCalledProcessError\u001b[0m                        Traceback (most recent call last)","\u001b[0;32m<ipython-input-19-7647a017a6bf>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mget_ipython\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrun_cell_magic\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'bash'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m''\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'./admixture -P ancient2merge_cleaned_nopug10KSnps.bed 2\\n'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m","\u001b[0;32m~/miniconda3/lib/python3.8/site-packages/IPython/core/interactiveshell.py\u001b[0m in \u001b[0;36mrun_cell_magic\u001b[0;34m(self, magic_name, line, cell)\u001b[0m\n\u001b[1;32m   2389\u001b[0m             \u001b[0;32mwith\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbuiltin_trap\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2390\u001b[0m                 \u001b[0margs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mmagic_arg_s\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2391\u001b[0;31m                 \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2392\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2393\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m~/miniconda3/lib/python3.8/site-packages/IPython/core/magics/script.py\u001b[0m in \u001b[0;36mnamed_script_magic\u001b[0;34m(line, cell)\u001b[0m\n\u001b[1;32m    140\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    141\u001b[0m                 \u001b[0mline\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mscript\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 142\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshebang\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mline\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    143\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    144\u001b[0m         \u001b[0;31m# write a basic docstring:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m<decorator-gen-103>\u001b[0m in \u001b[0;36mshebang\u001b[0;34m(self, line, cell)\u001b[0m\n","\u001b[0;32m~/miniconda3/lib/python3.8/site-packages/IPython/core/magic.py\u001b[0m in \u001b[0;36m<lambda>\u001b[0;34m(f, *a, **k)\u001b[0m\n\u001b[1;32m    185\u001b[0m     \u001b[0;31m# but it's overkill for just that one bit of state.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    186\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mmagic_deco\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 187\u001b[0;31m         \u001b[0mcall\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mlambda\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    188\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    189\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mcallable\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m~/miniconda3/lib/python3.8/site-packages/IPython/core/magics/script.py\u001b[0m in \u001b[0;36mshebang\u001b[0;34m(self, line, cell)\u001b[0m\n\u001b[1;32m    243\u001b[0m             \u001b[0msys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstderr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mflush\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    244\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0margs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mraise_error\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreturncode\u001b[0m\u001b[0;34m!=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 245\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mCalledProcessError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreturncode\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moutput\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstderr\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0merr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    246\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    247\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_run_script\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mto_close\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;31mCalledProcessError\u001b[0m: Command 'b'./admixture -P ancient2merge_cleaned_nopug10KSnps.bed 2\\n'' returned non-zero exit status 255."]}]},{"metadata":{"trusted":false},"id":"continuous-planning","cell_type":"code","source":"%%bash\n./plink --bfile ancient2merge_cleaned_nopug10KSnps --geno .99 --make-bed --out ancient2merge_cleaned_nopug10KSnpsNoMissing ","execution_count":20,"outputs":[{"name":"stdout","output_type":"stream","text":"PLINK v1.90b6.22 64-bit (16 Apr 2021)          www.cog-genomics.org/plink/1.9/\n(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\nLogging to ancient2merge_cleaned_nopug10KSnpsNoMissing.log.\nOptions in effect:\n  --bfile ancient2merge_cleaned_nopug10KSnps\n  --geno .99\n  --make-bed\n  --out ancient2merge_cleaned_nopug10KSnpsNoMissing\n\n9409 MB RAM detected; reserving 4704 MB for main workspace.\n10000 variants loaded from .bim file.\n92 people (51 males, 41 females) loaded from .fam.\n92 phenotype values loaded from .fam.\nUsing 1 thread (no multithreaded calculations invoked).\nBefore main variant filters, 92 founders and 0 nonfounders present.\nCalculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\nTotal genotyping rate is 0.644013.\n15 variants removed due to missing genotype data (--geno).\n9985 variants and 92 people pass filters and QC.\nAmong remaining phenotypes, 0 are cases and 92 are controls.\n--make-bed to ancient2merge_cleaned_nopug10KSnpsNoMissing.bed +\nancient2merge_cleaned_nopug10KSnpsNoMissing.bim +\nancient2merge_cleaned_nopug10KSnpsNoMissing.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"}]},{"metadata":{"trusted":false},"id":"authorized-death","cell_type":"code","source":"%%bash\n./plink --bfile GreSalAndModernEurasia10kSnps --extract ancient2merge_cleaned_nopug10KSnpsNoMissing.bim --make-bed --out GreSalAndModernEurasia10kSnpsNoMissing","execution_count":21,"outputs":[{"name":"stdout","output_type":"stream","text":"PLINK v1.90b6.22 64-bit (16 Apr 2021)          www.cog-genomics.org/plink/1.9/\n(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\nLogging to GreSalAndModernEurasia10kSnpsNoMissing.log.\nOptions in effect:\n  --bfile GreSalAndModernEurasia10kSnps\n  --extract ancient2merge_cleaned_nopug10KSnpsNoMissing.bim\n  --make-bed\n  --out GreSalAndModernEurasia10kSnpsNoMissing\n\n9409 MB RAM detected; reserving 4704 MB for main workspace.\n10000 variants loaded from .bim file.\n1231 people (654 males, 300 females, 277 ambiguous) loaded from .fam.\nAmbiguous sex IDs written to GreSalAndModernEurasia10kSnpsNoMissing.nosex .\n476 phenotype values loaded from .fam.\n--extract: 9985 variants remaining.\nUsing 1 thread (no multithreaded calculations invoked).\nBefore main variant filters, 1231 founders and 0 nonfounders present.\nCalculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\nTotal genotyping rate is 0.923415.\n9985 variants and 1231 people pass filters and QC.\nAmong remaining phenotypes, 51 are cases and 425 are controls.  (755 phenotypes\nare missing.)\n--make-bed to GreSalAndModernEurasia10kSnpsNoMissing.bed +\nGreSalAndModernEurasia10kSnpsNoMissing.bim +\nGreSalAndModernEurasia10kSnpsNoMissing.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"},{"name":"stderr","output_type":"stream","text":"Warning: Ignoring phenotypes of missing-sex samples.  If you don't want those\nphenotypes to be ignored, use the --allow-no-sex flag.\n"}]},{"metadata":{"trusted":false},"id":"killing-georgia","cell_type":"code","source":"%%bash\n./admixture GreSalAndModernEurasia10kSnpsNoMissing.bed 2","execution_count":22,"outputs":[{"name":"stdout","output_type":"stream","text":"****                   ADMIXTURE Version 1.3.0                  ****\n****                    Copyright 2008-2015                     ****\n****           David Alexander, Suyash Shringarpure,            ****\n****                John  Novembre, Ken Lange                   ****\n****                                                            ****\n****                 Please cite our paper!                     ****\n****   Information at www.genetics.ucla.edu/software/admixture  ****\n\nRandom seed: 43\nPoint estimation method: Block relaxation algorithm\nConvergence acceleration algorithm: QuasiNewton, 3 secant conditions\nPoint estimation will terminate when objective function delta < 0.0001\nEstimation of standard errors disabled; will compute point estimates only.\nSize of G: 1231x9985\nPerforming five EM steps to prime main algorithm\n1 (EM) \tElapsed: 1.098\tLoglikelihood: -1.24786e+07\t(delta): 7.10761e+06\n2 (EM) \tElapsed: 1.099\tLoglikelihood: -1.21472e+07\t(delta): 331438\n3 (EM) \tElapsed: 1.146\tLoglikelihood: -1.20521e+07\t(delta): 95088.4\n4 (EM) \tElapsed: 1.172\tLoglikelihood: -1.2006e+07\t(delta): 46131.3\n5 (EM) \tElapsed: 1.052\tLoglikelihood: -1.19819e+07\t(delta): 24029.5\nInitial loglikelihood: -1.19819e+07\nStarting main algorithm\n1 (QN/Block) \tElapsed: 2.632\tLoglikelihood: -1.19505e+07\t(delta): 31456.8\n2 (QN/Block) \tElapsed: 3.065\tLoglikelihood: -1.19369e+07\t(delta): 13557.8\n3 (QN/Block) \tElapsed: 3.735\tLoglikelihood: -1.19311e+07\t(delta): 5794.18\n4 (QN/Block) \tElapsed: 3.83\tLoglikelihood: -1.19275e+07\t(delta): 3675.65\n5 (QN/Block) \tElapsed: 4.405\tLoglikelihood: -1.19256e+07\t(delta): 1805.1\n6 (QN/Block) \tElapsed: 4.526\tLoglikelihood: -1.19242e+07\t(delta): 1463.91\n7 (QN/Block) \tElapsed: 4.894\tLoglikelihood: -1.19228e+07\t(delta): 1427.1\n8 (QN/Block) \tElapsed: 5.019\tLoglikelihood: -1.19215e+07\t(delta): 1283.98\n9 (QN/Block) \tElapsed: 5.08\tLoglikelihood: -1.19205e+07\t(delta): 952.701\n10 (QN/Block) \tElapsed: 4.161\tLoglikelihood: -1.192e+07\t(delta): 527.487\n11 (QN/Block) \tElapsed: 4.011\tLoglikelihood: -1.19197e+07\t(delta): 252.685\n12 (QN/Block) \tElapsed: 3.927\tLoglikelihood: -1.19195e+07\t(delta): 194.273\n13 (QN/Block) \tElapsed: 3.957\tLoglikelihood: -1.19192e+07\t(delta): 345.138\n14 (QN/Block) \tElapsed: 3.881\tLoglikelihood: -1.19187e+07\t(delta): 543.008\n15 (QN/Block) \tElapsed: 3.945\tLoglikelihood: -1.1918e+07\t(delta): 621.386\n16 (QN/Block) \tElapsed: 3.489\tLoglikelihood: -1.19177e+07\t(delta): 298.157\n17 (QN/Block) \tElapsed: 3.303\tLoglikelihood: -1.19176e+07\t(delta): 147.82\n18 (QN/Block) \tElapsed: 3.412\tLoglikelihood: -1.19175e+07\t(delta): 125.814\n19 (QN/Block) \tElapsed: 3.209\tLoglikelihood: -1.19173e+07\t(delta): 135.006\n20 (QN/Block) \tElapsed: 3.364\tLoglikelihood: -1.19172e+07\t(delta): 171.383\n21 (QN/Block) \tElapsed: 3.19\tLoglikelihood: -1.1917e+07\t(delta): 167.101\n22 (QN/Block) \tElapsed: 3.2\tLoglikelihood: -1.19169e+07\t(delta): 133.535\n23 (QN/Block) \tElapsed: 3.455\tLoglikelihood: -1.19167e+07\t(delta): 114.355\n24 (QN/Block) \tElapsed: 3.58\tLoglikelihood: -1.19166e+07\t(delta): 115.532\n25 (QN/Block) \tElapsed: 3.246\tLoglikelihood: -1.19164e+07\t(delta): 227.229\n26 (QN/Block) \tElapsed: 3.114\tLoglikelihood: -1.19162e+07\t(delta): 171.376\n27 (QN/Block) \tElapsed: 3.248\tLoglikelihood: -1.19161e+07\t(delta): 126.522\n28 (QN/Block) \tElapsed: 3.21\tLoglikelihood: -1.19161e+07\t(delta): 45.2705\n29 (QN/Block) \tElapsed: 3.179\tLoglikelihood: -1.19157e+07\t(delta): 313.917\n30 (QN/Block) \tElapsed: 3.39\tLoglikelihood: -1.19156e+07\t(delta): 172.692\n31 (QN/Block) \tElapsed: 3.66\tLoglikelihood: -1.19155e+07\t(delta): 118.922\n32 (QN/Block) \tElapsed: 3.136\tLoglikelihood: -1.19154e+07\t(delta): 32.1461\n33 (QN/Block) \tElapsed: 3.23\tLoglikelihood: -1.19153e+07\t(delta): 145.6\n34 (QN/Block) \tElapsed: 3.116\tLoglikelihood: -1.19152e+07\t(delta): 80.7757\n35 (QN/Block) \tElapsed: 3.511\tLoglikelihood: -1.19152e+07\t(delta): 38.6316\n36 (QN/Block) \tElapsed: 3.152\tLoglikelihood: -1.19151e+07\t(delta): 24.9723\n37 (QN/Block) \tElapsed: 3.153\tLoglikelihood: -1.19151e+07\t(delta): 11.5462\n38 (QN/Block) \tElapsed: 3.271\tLoglikelihood: -1.19151e+07\t(delta): 22.9112\n39 (QN/Block) \tElapsed: 3.135\tLoglikelihood: -1.19151e+07\t(delta): 12.6913\n40 (QN/Block) \tElapsed: 3.161\tLoglikelihood: -1.19151e+07\t(delta): 3.7571\n41 (QN/Block) \tElapsed: 3.15\tLoglikelihood: -1.19151e+07\t(delta): 2.39553\n42 (QN/Block) \tElapsed: 3.557\tLoglikelihood: -1.19151e+07\t(delta): 4.60672\n43 (QN/Block) \tElapsed: 3.697\tLoglikelihood: -1.19151e+07\t(delta): 2.22077\n44 (QN/Block) \tElapsed: 3.174\tLoglikelihood: -1.19151e+07\t(delta): 1.05943\n45 (QN/Block) \tElapsed: 3.223\tLoglikelihood: -1.19151e+07\t(delta): 0.389332\n46 (QN/Block) \tElapsed: 3.129\tLoglikelihood: -1.19151e+07\t(delta): 0.586008\n47 (QN/Block) \tElapsed: 3.139\tLoglikelihood: -1.19151e+07\t(delta): 0.571477\n48 (QN/Block) \tElapsed: 3.135\tLoglikelihood: -1.19151e+07\t(delta): 0.128897\n49 (QN/Block) \tElapsed: 3.133\tLoglikelihood: -1.19151e+07\t(delta): 0.0157161\n50 (QN/Block) \tElapsed: 3.161\tLoglikelihood: -1.19151e+07\t(delta): 0.00959872\n51 (QN/Block) \tElapsed: 3.129\tLoglikelihood: -1.19151e+07\t(delta): 0.01498\n52 (QN/Block) \tElapsed: 3.113\tLoglikelihood: -1.19151e+07\t(delta): 0.0012761\n53 (QN/Block) \tElapsed: 3.135\tLoglikelihood: -1.19151e+07\t(delta): 0.00020312\n54 (QN/Block) \tElapsed: 3.123\tLoglikelihood: -1.19151e+07\t(delta): 2.11261e-05\nSummary: \nConverged in 54 iterations (210.374 sec)\nLoglikelihood: -11915067.857171\nFst divergences between estimated populations: \n\tPop0\t\nPop0\t\nPop1\t0.143\t\nWriting output files.\n"}]},{"metadata":{"trusted":false},"id":"lyric-cambridge","cell_type":"code","source":"%%bash\ncp GreSalAndModernEurasia10kSnpsNoMissing.2.P ancient2merge_cleaned_nopug10KSnpsNoMissing.2.P.in ","execution_count":23,"outputs":[]},{"metadata":{"trusted":false},"id":"hazardous-monday","cell_type":"code","source":"%%bash\n./admixture -P ancient2merge_cleaned_nopug10KSnpsNoMissing.bed 2","execution_count":24,"outputs":[{"name":"stdout","output_type":"stream","text":"****                   ADMIXTURE Version 1.3.0                  ****\n****                    Copyright 2008-2015                     ****\n****           David Alexander, Suyash Shringarpure,            ****\n****                John  Novembre, Ken Lange                   ****\n****                                                            ****\n****                 Please cite our paper!                     ****\n****   Information at www.genetics.ucla.edu/software/admixture  ****\n\nWill read fixed P from files\nRandom seed: 43\nPoint estimation method: Block relaxation algorithm\nP parameters will be fixed at initial values.\nConvergence acceleration algorithm: QuasiNewton, 3 secant conditions\nPoint estimation will terminate when objective function delta < 0.0001\nEstimation of standard errors disabled; will compute point estimates only.\nInitializing parameter matrices from input files:\n  Read P input from ancient2merge_cleaned_nopug10KSnpsNoMissing.2.P.in\nSize of G: 92x9985\nPerforming five EM steps to prime main algorithm\n1 (EM) \tElapsed: 0.037\tLoglikelihood: -692052\t(delta): 11166.4\n2 (EM) \tElapsed: 0.043\tLoglikelihood: -682969\t(delta): 9082.46\n3 (EM) \tElapsed: 0.036\tLoglikelihood: -675141\t(delta): 7828.42\n4 (EM) \tElapsed: 0.035\tLoglikelihood: -668395\t(delta): 6745.93\n5 (EM) \tElapsed: 0.035\tLoglikelihood: -662620\t(delta): 5774.96\nInitial loglikelihood: -662620\nStarting main algorithm\n1 (QN/Block) \tElapsed: 0.091\tLoglikelihood: -629689\t(delta): 32931.4\n2 (QN/Block) \tElapsed: 0.092\tLoglikelihood: -629643\t(delta): 45.8976\n3 (QN/Block) \tElapsed: 0.182\tLoglikelihood: -629640\t(delta): 2.65325\n4 (QN/Block) \tElapsed: 0.177\tLoglikelihood: -629638\t(delta): 2.02786\n5 (QN/Block) \tElapsed: 0.178\tLoglikelihood: -629638\t(delta): 0.381344\n6 (QN/Block) \tElapsed: 0.182\tLoglikelihood: -629638\t(delta): 0.000215273\n7 (QN/Block) \tElapsed: 0.181\tLoglikelihood: -629638\t(delta): 1.5134e-09\nSummary: \nConverged in 7 iterations (1.538 sec)\nLoglikelihood: -629637.628283\nFst divergences between estimated populations: \n\tPop0\t\nPop0\t\nPop1\t0.143\t\nWriting output files.\n"}]},{"metadata":{"trusted":false},"id":"tropical-survey","cell_type":"code","source":"","execution_count":null,"outputs":[]}],"metadata":{"kernelspec":{"name":"python3","display_name":"Python 3","language":"python"},"language_info":{"name":"python","version":"3.7.10","mimetype":"text/x-python","codemirror_mode":{"name":"ipython","version":3},"pygments_lexer":"ipython3","nbconvert_exporter":"python","file_extension":".py"}},"nbformat":4,"nbformat_minor":5}